### PR TITLE
Use Python Version: let user select architecture as an advanced option

### PR DESCRIPTION
**Pull request by brcrista**
The hosted agents only have 64-bit versions of Python installed, but users are able to install 32-bit versions on private agents.  They can also install 32-bit versions on hosted Windows agents from NuGet using the 3rd party "Install Python" task.

Since hosted agents only have 64-bit, I am putting this setting under the "Advanced" tab with a default value of x64.  When the task fails to find a version, it will now list versions _with architectures_ available to the agent.

Testing:
- L0
- Manually with private Windows agent

### CODE REVIEW

File: Tasks/UsePythonVersion/Tests/L0.ts [hunk 1]
```diff
@@ -119,14 +121,52 @@ describe('UsePythonVersion L0 Suite', function () {
...
+                '2.7.13 (x64)'
             ].join(EOL);
 
             assert.strictEqual(e.message, expectedMessage);
             done();
         }
     });
 
+    it('selects architecture passed as input', async function () {
+        let buildVariables: { [key: string]: string } = {};
```
Comment (lkillgore) [inline Tasks/UsePythonVersion/Tests/L0.ts]: It looks like you can make this `const`.

Comment (brcrista) [reply Tasks/UsePythonVersion/Tests/L0.ts]: Yeah, I've come around to your opinion on this.  I'll change this next time I touch this file.

File: Tasks/UsePythonVersion/task.json [hunk 2]
```diff
@@ -34,6 +41,19 @@
...
+        },
+        {
+            "name": "architecture",
+            "type": "pickList",
+            "label": "Architecture",
+            "defaultValue": "x64",
+            "required": true,
+            "helpMarkDown": "The target architecture (x86, x64) of the Python interpeter.",
+            "groupName": "advanced",
+            "options": {
```
Comment (jpricketMSFT) [inline Tasks/UsePythonVersion/task.json]: I am not sure if this is appropriate here, but in most cases with options like this we have a "default" option that is selected automatically and allows the Task to make the most appropriate choice. I.e. choose the one that matches the OS arch.

Comment (brcrista) [reply Tasks/UsePythonVersion/task.json]: See a few lines above, the default is x64.  The agent's architecture won't be known at build definition time, but the hosted agents only have x64 Pythons.

Comment (madhurig) [reply Tasks/UsePythonVersion/task.json]: We have the "default" on several picklists (e.g. Java version) to make it just work on any agent but it does cause a lot of confusion. In this case not having "default" is better.

Comment (jpricketMSFT) [reply Tasks/UsePythonVersion/task.json]: Yes that is basically what I was thinking, but as Madhuri says, it can cause a lot of confusion. This is better.

**Pull request merged**